include/pisa/scorer/bm25.hpp

```diff
@@ -1,30 +1,43 @@
 #pragma once
 
-#include <cstdint>
-#include <cmath>
 #include <algorithm>
+#include <cmath>
+#include <cstdint>
+
+#include "index_scorer.hpp"
 namespace pisa {
 
-    struct bm25 {
-        static constexpr float b = 0.4;
-        static constexpr float k1 = 0.9;
+template <typename Wand>
+struct bm25 : public index_scorer<Wand> {
+    static constexpr float b = 0.4;
+    static constexpr float k1 = 0.9;
 
+    using index_scorer<Wand>::index_scorer;
 
-        static float doc_term_weight(uint64_t freq, float norm_len)
-        {
-            float f = (float)freq;
-            return f / (f + k1 * (1.0f - b + b * norm_len));
-        }
+    static float doc_term_weight(uint64_t freq, float norm_len)
+    {
+        float f = (float)freq;
+        return f / (f + k1 * (1.0f - b + b * norm_len));
+    }
 
-        //IDF (inverse document frequency)
-        static float query_term_weight(uint64_t freq, uint64_t df, uint64_t num_docs)
-        {
-            float f = (float)freq;
-            float fdf = (float)df;
-            float idf = std::log((float(num_docs) - fdf + 0.5f) / (fdf + 0.5f));
-            static const float epsilon_score = 1.0E-6;
-            return f * std::max(epsilon_score, idf) * (1.0f + k1);
```

Comment:
Where did this `f *` go? Am I missing something or has the scorer changed?

Reply:
Nevermind, I see what you have done. This is the query term weight, and you do this multiplication when you create the cursor.